src/internal/schemas/schema.go

```diff
@@ -1,10 +1,53 @@
 package schemas
 
-import "embed"
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"strings"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+)
 
 //go:embed *.json
 var Schemas embed.FS
 
 const (
 	SCHEMA_PREFIX = "oscal_complete_schema-"
+	SCHEMA_PATH   = "https://github.com/defenseunicorns/go-oscal/tree/main/src/internal/schemas"
```

Comment:
Other Ideas:
- use the `$id` from the schema to keep current behavior (dead link)
- point to the json reference on the nist website for the current schema verison

Reply:
What are the pro/cons associated here - dead link behavior sounds non-optimal but I am looking to understand what that currently means. 

What does this mean for the `compile()` operation? Does it require connectivity to retrieve data/metadata?

Reply:
It can be any path that identifies the schema source of truth, but it doesn't fetch the resources, it is used for the error documentation as far as I can tell. I just did some refactoring that may help clear up the confusion, but essentialy there is no `CompileString` method anymore, so now I use `AddResource` to essentially do the exact same thing. 

afaik `jsonschema` doesn't have the logic to fetch, but allows for overidding the `Load` method which would allow for fetching if the implementer desired it. 

essentially its the filepath, but we need the schema to always be retrievable in airgap, so we have to embed, and embedding doesn't have a nice path. 

this used to be the `$id` in the schema, but that doesn't lead anywhere http://csrc.nist.gov/ns/oscal/1.0/1.0.4/oscal-complete-schema.json

Reply:
currently would output in a `ValidatorError` like this `https://github.com/defenseunicorns/go-oscal/tree/main/src/internal/schemas/oscal_complete_schema-1-0-4.json#/definitions/oscal-complete-oscal-metadata:party/properties/uuid/pattern`

Reply:
previously: `http://csrc.nist.gov/ns/oscal/1.0/1.0.4/oscal-complete-schema.json#/definitions/oscal-complete-oscal-metadata:party/properties/uuid/pattern`

Reply:
Appreciate the breakdown - this makes a lot of sense and it looks like you accounted for connectivity constraints. The default you have established feels like a sound choice.